atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java

```diff
@@ -59,36 +58,35 @@
 import com.palantir.atlasdb.keyvalue.api.RowResult;
 import com.palantir.atlasdb.keyvalue.api.TableReference;
 import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.keyvalue.dbkvs.DbKeyValueServiceConfiguration;
 import com.palantir.atlasdb.keyvalue.dbkvs.impl.ranges.DbKvsGetRanges;
 import com.palantir.atlasdb.keyvalue.impl.AbstractKeyValueService;
 import com.palantir.atlasdb.keyvalue.impl.Cells;
 import com.palantir.common.base.ClosableIterator;
 import com.palantir.common.base.ClosableIterators;
 import com.palantir.common.base.Throwables;
-import com.palantir.db.oracle.JdbcHandler;
 import com.palantir.exception.PalantirSqlException;
 import com.palantir.nexus.db.sql.AgnosticLightResultRow;
 import com.palantir.nexus.db.sql.AgnosticResultRow;
 import com.palantir.nexus.db.sql.AgnosticResultSet;
-import com.palantir.nexus.db.sql.PalantirSqlConnection;
+import com.palantir.nexus.db.sql.SqlConnection;
 import com.palantir.util.paging.AbstractPagingIterable;
 import com.palantir.util.paging.SimpleTokenBackedResultsPage;
 import com.palantir.util.paging.TokenBackedBasicResultsPage;
 
 public class DbKvs extends AbstractKeyValueService {
     private static final Logger log = LoggerFactory.getLogger(DbKvs.class);
+    private final DbKeyValueServiceConfiguration config;
     private final DbTableFactory dbTables;
-    private final PalantirSqlConnectionSupplier connections;
-    private final JdbcHandler jdbcHandler;
+    private final SqlConnectionSupplier connections;
 
-    public DbKvs(ExecutorService executor,
+    public DbKvs(DbKeyValueServiceConfiguration config,
                  DbTableFactory dbTables,
-                 PalantirSqlConnectionSupplier connections,
-                 JdbcHandler jdbcHandler) {
-        super(executor);
+                 SqlConnectionSupplier connections) {
+        super(AbstractKeyValueService.createFixedThreadPool("Atlas Relational KVS", config.poolSize()));
```

Comment:
Any reason we would want to be able to supply this externally, as is done originally?